sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5418,12 +5421,28 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
     // No lock is needed here, unlike the immediate commandlist case above,
     // because the kernels are not actually submitted yet. Kernels will be
     // submitted only when the comamndlist is closed. Then, a lock is held.
-    ZE_CALL(zeCommandListAppendLaunchKernel,
+    zeResult = ZE_CALL_NOCHECK(zeCommandListAppendLaunchKernel,
             (CommandList->first, Kernel->ZeKernel, &ZeThreadGroupDimensions,
              ZeEvent, (*Event)->WaitList.Length,
              (*Event)->WaitList.ZeEventList));
   }
 
+  if(zeResult != ZE_RESULT_SUCCESS){
+	uint32_t sourceAttributesLength = 0U;
+	ZE_CALL(zeKernelGetSourceAttributes, (Kernel->ZeKernel, &sourceAttributesLength, nullptr));
+	char *sourceAttributes = new char[sourceAttributesLength];
+	ZE_CALL(zeKernelGetSourceAttributes, (Kernel->ZeKernel, &sourceAttributesLength, &sourceAttributes));
+	const std::string attributesString(sourceAttributes, sourceAttributes + sourceAttributesLength);
+	delete[] sourceAttributes;
+
+	const std::regex invalidKernelAttributeRegex("invalid_kernel\\(.*\\)");
+	std::smatch invalidKernelAttributeMatch;
+	if(std::regex_search(attributesString, invalidKernelAttributeMatch, invalidKernelAttributeRegex)){
+		throw std::invalid_argument(invalidKernelAttributeMatch[0].str());
```

Comment:
The plugin is not supposed to throw exceptions, instead an error code should be returned to SYCL RT. The easiest is to use PI_ERROR_PLUGIN_SPECIFIC_ERROR, see existing uses

Reply:
sure